test/ossl_shim/ossl_shim.cc

```diff
@@ -393,8 +393,10 @@ static int TicketKeyCallback(SSL *ssl, uint8_t *key_name, uint8_t *iv,
     return 0;
   }
 
-  *p++ = OSSL_PARAM_construct_utf8_string(OSSL_MAC_PARAM_DIGEST, "SHA256", 0);
-  *p++ = OSSL_PARAM_construct_octet_string(OSSL_MAC_PARAM_KEY, kZeros,
+  *p++ = OSSL_PARAM_construct_utf8_string(OSSL_MAC_PARAM_DIGEST,
+                                          const_cast<char *>("SHA256"), 0);
+  *p++ = OSSL_PARAM_construct_octet_string(OSSL_MAC_PARAM_KEY,
+                                           const_cast<void *>(kZeros),
```

Comment:
You need to change this API to use const char *, and const void*,
Otherwise this API is unusable from C++

Reply:
We can't change this API like that.  When constructing params, we've no idea if they will be read or written.  Thus the pointers passed in cannot be _const_.

Reply:
Hmm, could this be actually API deficiency? Should we have also read-only params?

Reply:
We can do read only params, we can't use the construct calls to make them.  To use the helpers, casts need to be introduced.

Adding a duplicate set of `OSSL_PARAM_construct_read_only_XXX` calls would be a possibility.  It's a lot of calls and they'd need to cast internally.

Reply:
Yes, I understand that we would have to have duplicate helper calls and I do not know whether it is worth it. I am just thinking as the read-only params would be quite common and the use of them with const char arrays would be also fairly common, whether this is something we should have or not. I suppose the construct calls could be added later, however implementing the read-only flag in the OSSL_PARAM would probably need to be done in 3.0 already. @levitte @mattcaswell what do you think? Is it worth the hassle?